File: rb_py_node/nodegrader/grader.js

// javascript grader app
const readline = require('readline')
const Grader = require('./graderClass')

const read = readline.Interface({
    input: process.stdin,
    output: process.stdout

})

// console stream input and output
read.question("Hello student, what is your name ", (name) => {
    
   read.question("On what topic was your test about? ", (assignment) => {
    
       read.question("Can you please enter your score? ", (grade) => {
        let num = parseInt(grade)
      
        const student = new Grader(name, assignment, num)
        student.printResults()
        // closing stream
        read.close()
       })
   })
})

File: rb_py_node/rubygrader/grader.rb
#grader app
require './graderClass.rb'

puts "Hello student, what is your name? "
name = gets

puts "What was your test about? "
assignment = gets 

grade = "".to_i
puts grade
# checking for valid input
while grade == 0
    puts "please enter a grade score from 0 to 100"
    grade = gets.to_i
    if grade < 0 && grade > 100
        grade = ""

    end
end

# initializing and printing
student = Grader.new(name, assignment, grade)
student.printResults()

